app/helpers/es.py

```diff
@@ -325,6 +325,23 @@ def add_update_bulk_action(self, document):
         }
         self.add_bulk_action(action)
 
+    def add_remove_outlier_bulk_action(self, document):
+        action = {
+            '_op_type': 'update',
+            '_index': document["_index"],
+            '_type': document["_type"],
+            '_id': document["_id"],
+            'retry_on_conflict': 10,
+            '_source': {
+                "script": {
+                    "source": "ctx._source.remove(\"outliers\"); " +
+                              "ctx._source.tags.remove(ctx._source.tags.indexOf(\"outlier\"))",
```

Comment:
I think this'll give errors if the "tags" field doesn't exist in the document. Under normal circumstances I don't think this state should ever occur but it's maybe best to have an if around it should to be sure

Reply:
Yes I have seen this error.
Normaly, here we only will have document that have "outliers" fields AND "tags" outlier.  But add a condition is a good idea.